pkg/render/typha.go

```diff
@@ -342,6 +342,7 @@ func (c *typhaComponent) typhaDeployment() *apps.Deployment {
 				},
 				Spec: v1.PodSpec{
 					Tolerations:                   c.tolerations(),
+					Affinity:                      toAffinity(c.installation.TyphaAffinity),
```

Comment:
Is there a better way to convert from our custom `TyphaAffinity` type to the near-identical `Affinity` type? I have all the subfields with matching names and pointer types, but am not able to cast it. I think this is because you cannot cast unless the structs are identical, and in this case our struct is a subset of the other.

If that's true, I was considering adding a `ToAffinity` receiver method off of the `TyphaAffinity` struct. But I haven't seen any other examples of receiver methods being added (besides the `DeepCopy` autogenerated methods.

Reply:
I think a helper like this is fine, you could do a receiver on our API type if you wanted. I think we've done that elsewhere in some of the elastic stuff maybe?

As for implementation, this seems fine